src/react-skycons.js

```diff
@@ -44,6 +44,9 @@ const ReactSkycons = React.createClass({
    this.state.skycons.set(React.findDOMNode(this), Skycons[nextProps.icon]);
   },
 
+  componentWillUnmount: function componentWillUnmount() {
+    this.state.skycons.remove(React.findDOMNode(this));
```

Comment:
Maybe you can call `this.state.skycons.pause()` before removing it? This method will call `cancelInteval`.